data/deduplicator/truncate/truncate_test.go

```diff
@@ -126,7 +125,7 @@ var _ = Describe("Truncate", func() {
 			})
 
 			It("returns false if the device id is empty", func() {
-				dataset.DeviceID = StringAsPointer("")
+				dataset.DeviceID = app.StringAsPointer("")
```

Comment:
No, a pointer to an empty string is not the same as nil. You can dereference a pointer to an empty string and get the string. If you dereference a nil, then you panic.